### PR TITLE
Move the testbed federation config to the production repositories #412

### PR DESCRIPTION
**Pull request by JuanPTM**
Add Keystone config file, modifications to horizon local settings to use federation and multiple ansible parameters for configuration. 

Part of https://github.com/SovereignCloudStack/issues/issues/412

### CODE REVIEW

**Comment by osfrickler** [comment]
recheck

File: {{cookiecutter.project_name}}/environments/kolla/configuration.yml [hunk 1]
```diff
@@ -62,4 +62,21 @@ ceph_public_network: {{cookiecutter.ceph_network_frontend}}
...
 ##########################################################
+# Keystone
+keystone_identity_providers:
+  - name: "keycloak"
+    openstack_domain: "keycloak"
+    protocol: "openid"
+    identifier: "https://keycloak.testbed.osism.xyz/auth/realms/osism"
+    public_name: "Authenticate via Keycloak"
+    attribute_mapping: "mappingId1"
+    metadata_folder: "{{ node_custom_config }}/keystone/federation/oidc/metadata"
```
Comment (berendt) [inline {{cookiecutter.project_name}}/environments/kolla/configuration.yml:74]: RAW blocks are required here to not parse {{ node_custom_config }}.

Have a look at https://github.com/osism/cfg-cookiecutter/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/environments/manager/configuration.yml#L30-L61 for aample. It's fine to have a single RAW block around all keystone parameters here.

**Review by berendt — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_name}}/environments/kolla/configuration.yml (hunk 1 above)
Comment (JuanPTM) [reply {{cookiecutter.project_name}}/environments/kolla/configuration.yml:74]: Added RAW block :+1:

**Comment by berendt** [comment]
One more change is required. Add the ``environments/kolla/files/overlays/keystone/wsgi-keystone.conf`` file to the ``_copy_without_render`` section in the ``cookiecutter.json`` file.

```
    "_copy_without_render": [
        "environments/kolla/files/overlays/keystone/wsgi-keystone.conf",
        "environments/kolla/files/overlays/haproxy/haproxy.cfg.*",
        "environments/kolla/files/overlays/neutron/ml2_conf.ini"
    ]
```

File: {{cookiecutter.project_name}}/environments/custom/configuration.yml [hunk 2]
```diff
@@ -1,4 +1,18 @@
 ---
-# Dummy variable to avoid error because ansible does not recognize the
-# file as a good configuration file when no variable in it.
-dummy:
+# keycloak-oidc-client-config
+
```
Comment (berendt) [inline {{cookiecutter.project_name}}/environments/custom/configuration.yml]: Looks like we have to use raw block here as well.

```
{%  raw -%}
```

File: {{cookiecutter.project_name}}/environments/custom/configuration.yml [hunk 3]
```diff
@@ -1,4 +1,18 @@
...
+  - "https://api.testbed.osism.xyz:5000/redirect_uri"
+  - "https://api.testbed.osism.xyz"
+  - "https://192.168.16.254:5000/redirect_uri"
+  - "https://192.168.16.254"
+
+keycloak_private_key_file_path: "{{ configuration_directory }}/environments/custom/files/keycloak/private_key.pem"
+keycloak_private_key_data: "{{ lookup('file', keycloak_private_key_file_path)| regex_replace('\n', '\\\\n') }}"
+
+keycloak_certificate_file_path: "{{ configuration_directory }}/environments/custom/files/keycloak/cert.crt"
+keycloak_certificate_data: "{{ lookup('file', keycloak_certificate_file_path)| regex_replace('\n', '\\\\n') }}"
```
Comment (berendt) [inline {{cookiecutter.project_name}}/environments/custom/configuration.yml:18]:
```
{%- endraw %}
```

**Review by berendt — CHANGES_REQUESTED** [review]

**Comment by JuanPTM** [comment]
Raw block added

**Pull request merged**